modules/local/euk_utils.nf

```diff
@@ -0,0 +1,88 @@
+process CONCATENATE_QUALITY_FILES {
+    tag "${meta.id}"
+
+    input:
+    tuple val(meta), path(input_files)
+    val output_name
+
+    output:
+    tuple val(meta), path("${meta.id}.${output_name}"), emit: concatenated_result
+
+    script:
+    """
+    echo "bin\tcompleteness\tcontamination\tncbi_lng" > "${meta.id}.${output_name}"
+    for i in ${input_files}; do
+        tail -n +2 \${i} > help_file
+        if [ -s help_file ]; then
+            grep -v "completeness" \${i} >> "${meta.id}.${output_name}"
+        fi
+    done
+    rm help_file
+    """
+}
+
+process MODIFY_QUALITY_FILE {
+
+    input:
+    path(quality_table_csv)
+    val output_name
+
+    output:
+    path("${output_name}"), emit: modified_result
+
+    script:
+    """
+    echo "genome,completeness,contamination" > ${output_name}
+    grep -v "completeness" ${quality_table_csv} | cut -f1-3 | tr '\t' ',' >> ${output_name} || true
+    """
+}
+
+process FILTER_QUALITY {
+    tag "${meta.id}"
+
+    label 'process_light'
+
+    input:
+    tuple val(meta), path(quality_file), path(concoct_bins, stageAs: "concoct_bins/*"), path(metabat_bins, stageAs: "metabat_bins/*"), path(concoct_bins_merged, stageAs: "concoct_bins_merged/*"), path(metabat_bins_merged, stageAs: "metabat_bins_merged/*")
+
+    output:
+    tuple val(meta), path("output_genomes/*"), path("quality_file.csv"), emit: qs50_filtered_genomes, optional: true
+    path "progress.log"                                       , emit: progress_log
+
+    script:
+    """
+    mkdir -p output_genomes
+    touch quality_file.csv
+
+    echo "Prepare drep quality"
+    # prepare drep quality file
+    grep -v "completeness" ${quality_file} |\
+    awk '{{if(\$2>=50 && \$2<=100 && \$3>=0 && \$3<=5){{print \$0}}}}' |\
```

Comment:
Is it possible to have completeness values > 100%

Reply:
hopefully not :D, the same as contamination < 0. I think that improves readability